tests/test_monitor_system.py

```diff
@@ -0,0 +1,126 @@
+import time
+from pathlib import Path
+
+from dvclive import Live
+from dvclive.monitor_system import MonitorCPU
+from dvclive.utils import parse_metrics
+
+
+def mock_psutil(mocker):
+    mocker.patch(
+        "dvclive.monitor_system.psutil.cpu_percent",
+        return_value=[10, 20, 30, 40, 50, 60],
+    )
+    mocker.patch("dvclive.monitor_system.psutil.cpu_count", return_value=6)
+
+    mocked_virtual_memory = mocker.MagicMock()
+    mocked_virtual_memory.percent = 20
+    mocked_virtual_memory.total = 4 * 1024**3
+
+    mocked_disk_usage = mocker.MagicMock()
+    mocked_disk_usage.used = 16 * 1024**3
+    mocked_disk_usage.percent = 50
+    mocked_disk_usage.total = 32 * 1024**3
+
+    mocking_dict = {
+        "virtual_memory": mocked_virtual_memory,
+        "disk_usage": mocked_disk_usage,
+    }
+    for function_name, return_value in mocking_dict.items():
+        mocker.patch(
+            f"dvclive.monitor_system.psutil.{function_name}",
+            return_value=return_value,
+        )
+
+
+def test_get_cpus_metrics_mocker(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        monitor = MonitorCPU(directories_to_monitor=["/", "/"])
+        monitor(live)
+        metrics = monitor._get_metrics()
+        monitor.end()
+
+    assert "system/cpu/usage (%)" in metrics
+    assert "system/cpu/count" in metrics
+    assert "system/cpu/parallelization (%)" in metrics
+    assert "system/ram/usage (%)" in metrics
+    assert "system/ram/usage (GB)" in metrics
+    assert "system/ram/total (GB)" in metrics
+    assert "system/disk/usage (%)/0" in metrics
+    assert "system/disk/usage (%)/1" in metrics
+    assert "system/disk/usage (GB)/0" in metrics
+    assert "system/disk/usage (GB)/1" in metrics
+    assert "system/disk/total (GB)/0" in metrics
+    assert "system/disk/total (GB)/1" in metrics
+
+
+def test_ignore_missing_directories(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        missing_directories = "______"
+        monitor = MonitorCPU(directories_to_monitor=["/", missing_directories])
+        monitor(live)
+        metrics = monitor._get_metrics()
+        monitor.end()
+
+    assert not Path(missing_directories).exists()
+
+    assert "system/disk/usage (%)/1" not in metrics
+    assert "system/disk/usage (GB)/1" not in metrics
+    assert "system/disk/total (GB)/1" not in metrics
+
+
+def test_monitor_system(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=True,
+    ) as live:
+        time.sleep(5 + 1)  # allow the thread to finish
```

Comment:
just some ideas how to make it more robust (we need more research on this)

- we can set a smaller interval
- then do while True with a small sleep inside and attempt to read the file, or check the Live class to see if there are new metrics already

the problem with the current approach primarily is that even after 6 seconds there is no guarantee, also it's a long time